internal/connector/internal/config/connectors.go

```diff
@@ -79,22 +148,29 @@ func (c *ConnectorsConfig) ReadFiles() error {
 			// Read the file
 			buf, err := os.ReadFile(path)
 			if err != nil {
-				err = gherrors.Errorf("error reading catalog file %s: %s", path, err)
-				return err
+				return gherrors.Errorf("error reading catalog file %s: %s", path, err)
 			}
 
 			entry := ConnectorCatalogEntry{}
 			err = json.Unmarshal(buf, &entry)
 			if err != nil {
-				err = gherrors.Errorf("error unmarshaling catalog file %s: %s", path, err)
-				return err
+				return gherrors.Errorf("error unmarshaling catalog file %s: %s", path, err)
+			}
+
+			// set catalog metadata from metadata config read earlier
+			id := entry.ConnectorType.Id
+			if meta, found := connectorMetadata[id]; found {
+				entry.ConnectorType.FeaturedRank = meta.FeaturedRank
+				entry.ConnectorType.Labels = meta.Labels
```

Comment:
should we merge labels from connector and the metadata ? there are some connectors that [have labels already ](https://github.com/bf2fc6cc711aee1a0c2a/cos-fleet-catalog-debezium/blob/270d14761e2a49c1e990f0f7a062f5a1efcc3c40/descriptors/mongodb/debezium-mongodb-1.9.6.Final.json#L10)

Reply:
Yeah, I considered that option. But, besides the code complexity in merging two lists (will it have to check for case/similar names?? to avoid duplicates) it would mean there are two sources of truth for labels. 
So I figured it's better to just make the catalog the source of non-metadata/core properties and make the metadata config file the exclusive source for the rest. 

It will also be easier to keep the labels consistent across in config files in one location rather than spreading them out over multiple pom files in the catalog. 

What do you think?

Reply:
It is fine with me

we just need to have some coordination as I don't know if the current labels were meant to be used so in such case we need to be sure they are added to the yaml file.

Reply:
You're right. So I'm going to generate the first version of the metadata file from the current catalog. That way we keep what we have right now. Also, for categories and internationalization, UI is going to use labels with the prefix `category-`, e.g. `category-aws`. That way they can ignore any other labels when displaying their list of categories.